language/move-prover/doc/user/spec-lang.md

```diff
@@ -206,6 +212,67 @@ The choice also comes in a form to select the *minimal* value from a set of inte
 choose min i: num where in_range(v, i) && v[i] == 2
 ```
 
+## Cast Operator
+
+In the specification language, we can use the same syntax `(e as T)` to cast an expression `e` with one integer type to
+`T`, an integer type of another size.
+
+
+## Shift Operator
+
+Shift operators `<<` and `>>` are supported in the specification language and both of them have the same semantics (including abort) with the Move language.
+
+## Bitwise Operators
+
+Move programs using bitwise operators `&`, `|` and `^` can be verified in the prover and these operators are also supported in the specification language.
+Due to encoding and efficiency issues, using bitwise operators has more caveats:
+
+- Integers involved in bitwise operations are encoded as `bv` types at the backend and two encodings of integers are not compatible. For instance,
+if a variable `v` involves in a bitwise operation such as `v & 2` or `v = a ^ b`, then when it is used in an arithmetic operation `v * w` or a shift operation `v << w`, `w` will be implicitly cast to a `bv` type in the Move program.
+However, the specification language does not support implicit type cast so users must explicitly use the builtin function `int2bv` in the specification: `v << int2bv(w)`.
+Not that since each `bv` type has a fixed length (from 8 to 256), values with type `num` cannot be converted into `bv`.
+
+
+- Verification of `bv` types is not efficient and may lead to timeout. As a result, users may prefer isolating bitwise operations from other operations and not using `int2bv`
```

Comment:
By the way, does `bv2int` also exist?

Reply:
Yes, but using `bv2int` will often lead to timeout so it is not encouraged to use it.

Reply:
Could you also mention `bv2int` and its limitation in the text?

Reply:
done